### DISCUSSION TITLE
Can I bind multisample texture to a shader variable?

### DISCUSSION THREAD

**Discussion by FlightBlaze** [DiligentGraphics/DiligentEngine#172] [Q&A] [answered]
Me: Trying to bind multisample texture to a shader variable
Diligent Engine:
```
Texture view 'Default SRV of texture 'Render Target Color'' bound to variable 'g_Background' is invalid:
single-sample texture is expected.
```
Me: (hanged himself)

**Question:**

Can I somehow bind multisample texture to a shader variable?

**Wherefore:**

I want to blend foreground and background with a custom blending mode in pixel shader

**Comment by TheMostDiligent** [comment, marked as answer]
You can only bind multisample texture to a special mutlisample texture variable type in the shader. You can't sample this texture with any sampler and can only fetch individual texels.
If yow want to use result of rendering with MSAA in a shader, you need to resolve the MS texture first into a single-sample texture and then bind it.
